src/elements/content-explorer/ItemGridThumbnail.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import * as React from 'react';
+import classNames from 'classnames';
+import getProp from 'lodash/get';
+import { getIcon } from '../common/item/iconCellRenderer';
+import './ItemGridThumbnail.scss';
+
+type Props = {
+    item: BoxItem,
+    token: Token,
+};
+
+const ItemGridThumbnail = ({ item, token }: Props) => {
+    let url;
+    const entries = getProp(item, 'representations.entries');
+    if (getProp(entries, '[0].status.state') === 'success') {
+        const entry = getProp(entries, '[0]');
+        const rep = getProp(entry, 'representation');
+        const template = getProp(entry, 'content.url_template');
+        if (token && template && rep) {
+            // Asset path for thumbnail is simply empty string.
+            const thumbnailBaseUrl = template.replace('{+asset_path}', rep === 'jpg' ? '' : '1.png');
+            url = `${thumbnailBaseUrl}?access_token=${token}`;
```

Comment:
It doesn't seem like `token` should be passed this deeply, since it's meant to be used for data access. Can the URL be passed in from a higher level component?